### PR TITLE
fix(tdigest): fix incorrect min/max default initialization

### PR DESCRIPTION
**Pull request by Tangruilin**
CentroidsWithDelta struct was default-initializing min and max to 0, which caused incorrect behavior when merging empty tdigests. Also delete Reset({}) overload to prevent accidental misuse.

Assisted-by: Claude Code:glm-5

### CODE REVIEW

File: src/types/tdigest.cc [hunk 1]
```diff
@@ -355,6 +355,7 @@ class TDigest {
   void Merge(const std::vector<TDigest>& others);
   void Add(const std::vector<double>& items);
   void Reset(const CentroidsWithDelta& centroid_list);
+  void Reset(std::initializer_list<CentroidsWithDelta>) = delete;
   void Reset();
   CentroidsWithDelta DumpCentroids() const;
```
[the remark this answers is not in the document]
Comment (jihuayu) [reply src/types/tdigest.cc:362]: I agree with Copilot. Could you please add the header file?

File: src/types/tdigest.cc [hunk 2]
```diff
@@ -355,6 +355,7 @@ class TDigest {
   void Merge(const std::vector<TDigest>& others);
   void Add(const std::vector<double>& items);
   void Reset(const CentroidsWithDelta& centroid_list);
+  void Reset(std::initializer_list<CentroidsWithDelta>) = delete;
```
Comment (jihuayu) [inline src/types/tdigest.cc:360]: Seeing this delete on its own is a bit confusing. Could you please add a comment explaining why it's being deleted?

File: src/types/tdigest.h [hunk 3]
```diff
@@ -47,10 +48,10 @@ struct Centroid {
 
 struct CentroidsWithDelta {
```
Comment (jihuayu) [inline src/types/tdigest.h:49]: Please add a regression test for merging only empty TDIGEST sources, then adding samples and verifying TDIGEST.MIN/MAX. This covers the externally observable behavior fixed by this PR.

**Review by jihuayu — COMMENTED** [review]
General LGTM. Thanks!

File: src/types/tdigest.cc (hunk 2 above)
Comment (Tangruilin) [reply src/types/tdigest.cc:360]: Done

File: src/types/tdigest.h (hunk 3 above)
Comment (Tangruilin) [reply src/types/tdigest.h:49]: Done
before:

Image

after:

Image

File: tests/cppunit/types/tdigest_test.cc [hunk 4]
```diff
@@ -948,3 +948,63 @@ TEST_F(RedisTDigestTest, MergeWithUserSpecifiedCompression) {
   // Verify total observations: dest(1) + src(1) = 2
   EXPECT_EQ(metadata.total_observations, 2);
 }
+
+TEST_F(RedisTDigestTest, MergeEmptySourcesThenAddSamples) {
```
Comment (jihuayu) [inline tests/cppunit/types/tdigest_test.cc]: I don’t think we need a C++ unit test here. Could you change it to a Go integration test instead? For command-level behavior like this, we generally prefer Go integration tests.

Comment (Tangruilin) [reply tests/cppunit/types/tdigest_test.cc]: Done

**Review by jihuayu — APPROVED** [review]

**Review by LindaSummer — APPROVED** [review]
Hi @Tangruilin ,

LGTM 🚀
Thanks very much for your contribution!
